lib/srv/desktop/windows_server.go

```diff
@@ -1401,3 +1406,20 @@ type otherName struct {
 type upn struct {
 	Value string `asn1:"utf8"`
 }
+
+// monitorErrorSender implements the io.StringWriter
+// interface in order to allow us to pass connection
+// monitor disconnect messages back to the frontend
+// over the tdp.Conn
+type monitorErrorSender struct {
+	log     logrus.FieldLogger
+	tdpConn *tdp.Conn
+}
+
+func (m *monitorErrorSender) WriteString(s string) (n int, err error) {
+	if err := m.tdpConn.SendError(s); err != nil {
+		m.log.Errorf("Failed to send TDP error message %v", err)
+	}
+
+	return 0, nil
```

Comment:
You're supposed to return the number of bytes that were written. Let's fix this hard-coded 0.

Reply:
e5c885e4ac2fe13d39a809aebfef9c4a5034e0f8